homeassistant/components/pushover/notify.py

```diff
@@ -47,12 +48,41 @@ def __init__(self, user_key, api_token):
     def send_message(self, message='', **kwargs):
         """Send a message to a user."""
         from pushover import RequestError
+        import re
 
         # Make a copy and use empty dict if necessary
         data = dict(kwargs.get(ATTR_DATA) or {})
 
         data['title'] = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
 
+        # Check for attachment.
+        if ATTR_ATTACHMENT in data:
+            # If attachment is a URL, use requests to open it as a stream.
+            if re.match('^[a-zA-Z]{3,5}://.+$', data[ATTR_ATTACHMENT]):
```

Comment:
This match also file URLs they need validate by `is_allowed_path`. Also this kind of regex without compiling is slow. use `data[].startswith("http")`. Sure, FTP will not work but how cares on 2020

Reply:
Fixed.